tests/test_pysoundfile.py

```diff
@@ -660,7 +665,7 @@ def test_seek_in_rplus_mode(sf_stereo_rplus):
 
 @pytest.mark.parametrize("use_default", [True, False])
 def test_truncate(file_stereo_rplus, use_default):
-    if isinstance(file_stereo_rplus, (str, int)):
+    if isinstance(file_stereo_rplus, (str, int, pathlib.Path)):
```

Comment:
This could be a problem when `pathlib` is not available.

Probably it's better to check `hasattr(file_stereo_rplus, '__fspath__')`?